src/Text/Pandoc/Writers/RST.hs

```diff
@@ -353,9 +353,20 @@ blockListToRST' :: PandocMonad m
                 -> [Block]       -- ^ List of block elements
                 -> RST m Doc
 blockListToRST' topLevel blocks = do
+  -- insert comment between list and quoted blocks, see #4248
+  let fixBlocks (b1:b2@(BlockQuote _):bs)
+        | isList b1 = b1 : commentSep : b2 : fixBlocks bs
+        where
+          isList (BulletList _)     = True
+          isList (OrderedList _ _)  = True
+          isList (DefinitionList _) = True
+          isList _                  = False
+          commentSep  = RawBlock "rst" ""
```

Comment:
Looking at the writer's code, I still don't see why this gives you `..`.
What happens if you use `RawBlock "rst" ".."`?

Reply:
Tests fail with a duplicated `..` line. Let's see whether i can find the code

Reply:
nope i can't find it. as far as i understand [this](https://github.com/jgm/pandoc/blob/master/src/Text/Pandoc/Writers/RST.hs#L236) is the line responsible for rendering a raw block.

maybe the `..` is actually added by [this line](https://github.com/jgm/pandoc/blob/master/src/Text/Pandoc/Writers/RST.hs#L368). if that's true i probably got the code wrong and have to understand that part of the rendering of a quoted block but it's evening here, i'll do tomorrow with a fresher mind

Reply:
inserting the comment separator before the quote seems to affect `stLastNested` so that `blockToRST'` adds [what looks like a comment](https://github.com/jgm/pandoc/blob/master/src/Text/Pandoc/Writers/RST.hs#L368) before the quote. [this commit](https://github.com/jgm/pandoc/commit/ca77f0a95e03cace027a235ebbc1effa99ea030a) connects this logic with the same problem we are trying to solve now. removing the empty `RawBlock`, my test fails. i have to understand how the status evolves

Reply:
basically [`blockToRST'`](https://github.com/jgm/pandoc/blob/master/src/Text/Pandoc/Writers/RST.hs#L362-L386) is a version of `fixBlocks` which uses status to terminate the previous block in order to avoid syntax errors. ideally we would have a single function going through the blocks to perform this check, but i'm having an hard time figuring out the meaning of `stLastNested` in order to unify the triggering conditions for an empty comment.

the logic around `stLastNested` seem to be contained mostly in `blockToRST'` linked above. the flag seems to be set most of the times but it's resetted for some specific blocks that we don't want to be followed by an empty comment, apparently.

i believe that the empty raw block before the quote leads to the flag being set and the comment to be added by `blockToRST'`, but i don't understand why lists block don't cause the same to happen.

i will try extending the test cases related to that code in order to get a better understanding while improving the software

Reply:
once understood the conditions within `blockToRST'` we could be able to make it stateless moving it to `fixBlocks`. it could be more readable and manageable

Reply:
i've not been able to understand why lists don't fall into the [default pattern match](https://github.com/jgm/pandoc/blob/master/src/Text/Pandoc/Writers/RST.hs#L379) into `blockToRST'`, if they did issue #4248 wouldn't exist. `RawBlocks` matched that pattern i believe, that's why adding a comment would turn out to be redundant in my test.